src/gui/tray/activitylistmodel.cpp

```diff
@@ -589,7 +593,7 @@ void ActivityListModel::combineActivityLists()
     _finalList.clear();
     endResetModel();
 
-    if (resultList.count() > 0) {
+    if (resultList.count() > 0 && _finalList.isEmpty()) {
```

Comment:
@claucambra Is this condition `_finalList.isEmpty()` needed? `_finalList.clear();` is always called before, so `_finalList` is empty by definition.
If it was not always empty, this would also mean that it would never get updated after getting populated for the first time.

Reply:
good catch, thanks

Reply:
Yes, but sometimes despite clearing the finalList will, somehow, not be cleared, making the rowCount increase by the number of new rows added. This gets caught by the model tester

Reply:
Ignore what I said, seems like the modeltester might be complaining about something else, going to take a look

Reply:
Okay, the whoel kerfuffle around this is now fixed :D